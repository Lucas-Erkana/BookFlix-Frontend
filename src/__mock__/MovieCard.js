import React from 'react';
import { BrowserRouter, Link } from 'react-router-dom';

const MovieCard = () => {
  const movie = [{
    name: 'Movie 1',
    image: 'https://res.cloudinary.com/dx0nauane/image/upload/v1624349663/Movie%20Card%20Images/Movie_1_1_xqjz3c.jpg',
    details: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla euismod, sapien vitae aliquam dapibus, magna quam ultricies nunc, nec tincidunt nisl nunc vitae elit. ',
    price: 100,
    id: 1,
  }];
  const rating = 5;

  return (
    <BrowserRouter>
      <div>
        <Link
          to={{
            pathname: `/movies/${movie.id}`,
          }}
          className="singlecard"
        >
          <img src={movie.image} alt={name} />
          <h4 className="bold-font movie-name">{movie.name}</h4>
          <hr className="dash" />
          <div className="movie-rating-price">
            <span className="bold-font gray-font">
              <i className="fa-sharp fa-solid fa-star-half-stroke" />
              {' '}
              {rating}
            </span>
            <span className="movie-circle-splitter" />
            <span className="bold-font gray-font">
              <i className="fa-solid fa-dollar-sign" />
              {' '}
              {movie.price}
            </span>
          </div>
          <p className="gray-font movie-details">{movie.details}</p>
        </Link>
      </div>
    </BrowserRouter>
  );
};

export default MovieCard;
