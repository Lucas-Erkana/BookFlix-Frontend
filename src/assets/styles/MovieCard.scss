.image-size-div {
  width: 20rem;
  overflow: hidden;

  .movie-image-size {
    width: 100%;
  }
}

.movie-name {
  color: black;
  text-decoration: none; /* Add this line to remove underline */
  text-decoration: underline;
  margin-top: 16px;
  background-color: rgba(255, 255, 255, 0.75);
  padding: 0 4px;
}

.movie-image {
  width: 100%;
  aspect-ratio: 1/1;
  object-fit: cover;
  object-position: top center;
}

.movie-name:hover {
  color: #98bf11;
}

.movie-list {
  display: flex;
  flex-wrap: wrap;
  padding: 3rem 0;
  justify-content: center;
}
