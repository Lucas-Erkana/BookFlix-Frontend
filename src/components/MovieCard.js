import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const ServiceCard = ({
  name, image, details, price, id,
}) => {
  const rating = Math.round((Math.random() * (5 - 4) + 4) * 10) / 10;

  return (
    <div>
      <Link
        to={{
          pathname: `/movies/${id}`,
        }}
        className="singlecard"
      >
        <img src={image} alt={name} className="movie-image" />
        <h4 className="bold-font movie-name">{name}</h4>
        <hr className="dash" />
        <div className="movie-rating-price">
          <span className="bold-font gray-font">
            <i className="fa-sharp fa-solid fa-star-half-stroke" />
            {' '}
            {rating}
          </span>
          <span className="movie-circle-splitter" />
          <span className="bold-font gray-font">
            <i className="fa-solid fa-dollar-sign" />
            {' '}
            {price}
          </span>
        </div>
        <p className="gray-font movie-details">{details}</p>
      </Link>
    </div>
  );
};

export default ServiceCard;

ServiceCard.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  details: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  id: PropTypes.number.isRequired,
};
