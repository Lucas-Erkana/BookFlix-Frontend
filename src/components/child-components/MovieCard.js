import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

const MovieCard = ({
  name, image, details, price, id,
}) => {
  const rating = Math.round((Math.random() * (5 - 4) + 4) * 10) / 10;

  return (
    <div>
      <div className="singlecard p-5">
        <NavLink to={{ pathname: `/movies/${id}` }}>
          <div className="image-size-div flex justify-center">
            <img src={image} alt={name} className="movie-image-size" />
          </div>
        </NavLink>
        <NavLink to={{ pathname: `/movies/${id}` }}>
          <h4 className="bold-font movie-name">{name}</h4>
        </NavLink>
        <hr className="dash" />
        <div className="movie-rating-price">
          <span className="bold-font gray-font">
            <i className="fa-sharp fa-solid fa-star-half-stroke" />
            {' '}
            {rating}
          </span>
          <span className="movie-circle-splitter" />
          <span className="bold-font gray-font">
            <i className="fa-solid fa-dollar-sign" />
            {' '}
            {price}
          </span>
        </div>
        <p className="gray-font movie-details">{details}</p>
      </div>
    </div>
  );
};

export default MovieCard;

MovieCard.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  details: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  id: PropTypes.number.isRequired,
};
