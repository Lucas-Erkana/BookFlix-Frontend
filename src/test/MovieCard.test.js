import { render, cleanup } from '@testing-library/react';
import MovieCard from '../__mock__/MovieCard';

afterEach(() => {
  cleanup();
});

describe('MovieCard', () => {
  it('MovieCard renders correctly', () => {
    const moviecard = render(<MovieCard />);
    expect(moviecard).toMatchSnapshot();
  });
});
