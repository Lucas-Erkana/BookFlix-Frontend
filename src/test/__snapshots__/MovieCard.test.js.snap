// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard MovieCard renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <a
          class="singlecard"
          href="/movies/undefined"
        >
          <img
            alt=""
          />
          <h4
            class="bold-font movie-name"
          />
          <hr
            class="dash"
          />
          <div
            class="movie-rating-price"
          >
            <span
              class="bold-font gray-font"
            >
              <i
                class="fa-sharp fa-solid fa-star-half-stroke"
              />
               
              5
            </span>
            <span
              class="movie-circle-splitter"
            />
            <span
              class="bold-font gray-font"
            >
              <i
                class="fa-solid fa-dollar-sign"
              />
               
            </span>
          </div>
          <p
            class="gray-font movie-details"
          />
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <a
        class="singlecard"
        href="/movies/undefined"
      >
        <img
          alt=""
        />
        <h4
          class="bold-font movie-name"
        />
        <hr
          class="dash"
        />
        <div
          class="movie-rating-price"
        >
          <span
            class="bold-font gray-font"
          >
            <i
              class="fa-sharp fa-solid fa-star-half-stroke"
            />
             
            5
          </span>
          <span
            class="movie-circle-splitter"
          />
          <span
            class="bold-font gray-font"
          >
            <i
              class="fa-solid fa-dollar-sign"
            />
             
          </span>
        </div>
        <p
          class="gray-font movie-details"
        />
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ServiceCard ServiceCard renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <a
          class="singlecard"
          href="/services/undefined"
        >
          <img
            alt=""
          />
          <h4
            class="bold-font service-name"
          />
          <hr
            class="dash"
          />
          <div
            class="service-rating-price"
          >
            <span
              class="bold-font gray-font"
            >
              <i
                class="fa-sharp fa-solid fa-star-half-stroke"
              />
               
              5
            </span>
            <span
              class="service-circle-splitter"
            />
            <span
              class="bold-font gray-font"
            >
              <i
                class="fa-solid fa-dollar-sign"
              />
               
            </span>
          </div>
          <p
            class="gray-font service-details"
          />
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <a
        class="singlecard"
        href="/services/undefined"
      >
        <img
          alt=""
        />
        <h4
          class="bold-font service-name"
        />
        <hr
          class="dash"
        />
        <div
          class="service-rating-price"
        >
          <span
            class="bold-font gray-font"
          >
            <i
              class="fa-sharp fa-solid fa-star-half-stroke"
            />
             
            5
          </span>
          <span
            class="service-circle-splitter"
          />
          <span
            class="bold-font gray-font"
          >
            <i
              class="fa-solid fa-dollar-sign"
            />
             
          </span>
        </div>
        <p
          class="gray-font service-details"
        />
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
